Fix dependency issue and add Snappy support

kruai opened the pull request:
# Overview

This PR is being created to address [ESSNTL-2022](https://issues.redhat.com/browse/ESSNTL-2022).
It fixes the dependency resolution error that was triggered by a conflict between connexion and v0.2 of openapi-schema-validator, which released recently. It also adds support for Snappy, so that when our Kafka brokers implement compression, we can decompress them.

## PR Checklist

- [x] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

kruai commented:
The main change I'm not sure about is the Dockerfile. I used `dnf module install` to match our approach with `postgresql`, but I'm not sure if that will work. If it doesn't, hopefully the standard `dnf install` will.

kruai commented:
Well, it looks like neither works. I might have to reach out to AppSRE to see what's available, or the Strimzi team to see how they implemented it on their end

thearifismail commented on Dockerfile:
```diff
@@ -4,6 +4,7 @@ USER root
 
 RUN dnf config-manager --disable rhel-8-for-x86_64-baseos-beta-rpms rhel-8-for-x86_64-appstream-beta-rpms
 RUN dnf module install -y postgresql:13
+RUN dnf install snappy-devel
```
Was any attempt made to build a local image?

kruai replied:
I answered when you asked in Slack, but: no, at this point, I had not. I ran this command on my own machine to get the Snappy installation

kruai commented:
Turns out we only need to install Snappy on our local machines in order to use python-snappy, so I removed the installation command from the Dockerfile. I'll also add that requirement to our README.

ryandillinfelton reviewed (approved):
LGTM! Thanks for tracking that down

ryandillinfelton commented:
> Turns out we only need to install Snappy on our local machines in order to use python-snappy, so I removed the installation command from the Dockerfile. I'll also add that requirement to our README.

Is that so? I sitll see Snappy installs on both Dockerfiles. I would expect it's necessary for deploys unless it's part of the base image but it' still in the dev one too. Am I missing something?

kruai commented:
> > Turns out we only need to install Snappy on our local machines in order to use python-snappy, so I removed the installation command from the Dockerfile. I'll also add that requirement to our README.
> 
> Is that so? I sitll see Snappy installs on both Dockerfiles. I would expect it's necessary for deploys unless it's part of the base image but it' still in the dev one too. Am I missing something?

I was trying out different permutations on the ephemeral nodes. I had removed both installation lines briefly to see if it would complain, and they didn't, but that could have been because they were already installed. I re-added them and it seems to work fine, so... fingers crossed that AppSRE doesn't have to add a module again!

ryandillinfelton commented:
> I was trying out different permutations on the ephemeral nodes. I had removed both installation lines briefly to see if it would complain, and they didn't, but that could have been because they were already installed. I re-added them and it seems to work fine, so... fingers crossed that AppSRE doesn't have to add a module again!

Shouldn't be an issue to reinstall a package so I see no issue with merging this as is.
